text/0000-project-descriptor-domains.md

```diff
@@ -0,0 +1,104 @@
+# Meta
+[meta]: #meta
+- Name: Project Descriptor Domains
+- Start Date: 2021-02-26
+- Author(s): hone
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: [PR#95](https://github.com/buildpacks/rfcs/pull/95)
+
+# Summary
+[summary]: #summary
+
+Change the Project Descriptor so all keys will live under a reverse domain namespace.
+
+# Motivation
+[motivation]: #motivation
+
+The current spec of the Project Descriptor is simple but restrictive for companies and other projects to adopt it for their own use case. The only level of non-spec'd customization is through the toml table `[metadata]` for arbitrary keys. This brings a lot of benefits and flexibility for the project. Unfortunately, this is fairly restrictive for anyone looking to extend the Project Descriptor for their own use case. Not being allowed to create your own "top level" keys, means non build concerns will be treated as "second class citizens". For instance, if you were using CNB for a FaaS product, it's plausible to have an `[event]` key to configure invocation that would be native to that platform. Forcing all this config to be under `[metadata]` is awkward.
+
+By not allowing this level of control, it limits the Project Descriptor's ability to be used in various configurations. In practice, Cloud Native Buildpacks only cares about defining a limited set of the schema.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The Project Descriptor will move to using reverse domains for top level keys as namespaces. This does two things:
+
+1. Allows any top level customization.
+1. Removes conflict of top level keys.
+
+The current schema as part of this proposal would look something like this:
+
+```TOML
```

Comment:
It might be useful to have an example of a `project.toml` that does NOT use buildpacks at all. Maybe a hypothetical `Dockerfile` example might look like. Or even a `[com.megasaasplatform]` that allows has a few example keys. Using nextjs/fly.io or similar as a basis maybe?

Reply:
@jabrown85 I added a fly.io example.